pkg/console/tlsconfig/tlsconfig_test.go

```diff
@@ -0,0 +1,228 @@
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"os"
+	"runtime"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	ocpconfigv1 "github.com/openshift/api/config/v1"
+	"k8s.io/client-go/util/cert"
+
+	"github.com/kubevirt/vm-console-proxy/api/v1alpha1"
+)
+
+var _ = Describe("TlsConfig", func() {
+	const (
+		certHostName = "unittest.test"
+	)
+
+	var (
+		tlsConfigPath string
+		certPath      string
+		keyPath       string
+
+		configWatch Watch
+	)
+
+	createTempFile := func(pattern string, content []byte) string {
+		tempFile, err := os.CreateTemp("", pattern)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(func() {
+			err := os.Remove(tempFile.Name())
+			if !errors.Is(err, os.ErrNotExist) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+		defer tempFile.Close()
+
+		_, err = tempFile.Write(content)
+		Expect(err).ToNot(HaveOccurred())
+
+		return tempFile.Name()
+	}
+
+	BeforeEach(func() {
+		tlsProfile := &v1alpha1.TlsSecurityProfile{
+			Type:         ocpconfigv1.TLSProfileIntermediateType,
+			Intermediate: &ocpconfigv1.IntermediateTLSProfile{},
+		}
+		tlsProfileJson, err := json.Marshal(tlsProfile)
```

Comment:
IMO yaml is more likely to be used by end users, so only use that?

Reply:
done